checkov/terraform/image_referencer/azure.py

```diff
@@ -0,0 +1,101 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from hcl2 import START_LINE, END_LINE
+
+from checkov.common.graph.graph_builder import CustomAttributes
+from checkov.common.images.image_referencer import Image
+from checkov.common.util.data_structures_utils import find_in_dict
+from checkov.common.util.type_forcers import force_list
+
+if TYPE_CHECKING:
+    from networkx import DiGraph
+
+SUPPORTED_AWS_IMAGE_RESOURCE_TYPES = {
+    "azurerm_batch_pool",
+    "azurerm_container_group",
+    "azurerm_linux_function_app",
+    "azurerm_linux_function_app_slot",
+    "azurerm_linux_web_app",
+    "azurerm_linux_web_app_slot",
+    "azurerm_spring_cloud_container_deployment",
+    "azurerm_windows_web_app",
+    "azurerm_windows_web_app_slot",
+}
+
+
+def extract_images_from_azure_resources(graph_connector: DiGraph) -> list[Image]:
+    images = []
+
+    resource_nodes = [
+        node
+        for node, resource_type in graph_connector.nodes(data=CustomAttributes.RESOURCE_TYPE)
+        if resource_type and resource_type in SUPPORTED_AWS_IMAGE_RESOURCE_TYPES
+    ]
+
+    supported_resources_graph = graph_connector.subgraph(resource_nodes)
+
+    for _, resource in supported_resources_graph.nodes(data=True):
+        image_names: list[str] = []
+        resource_type = resource[CustomAttributes.RESOURCE_TYPE]
+
+        if resource_type == "azurerm_batch_pool":
```

Comment:
@gruebel Perhaps a dict pattern, where for each resource type you get a callable, is a better pattern than the `elif` jungle you have here?